### ISSUE TITLE
Opening a dataset on Azure much slower than on S3

### ISSUE THREAD

**Issue by lsterzinger** [intake/filesystem_spec#681]
I'm trying to access NOAA GOES-16 satellite data on the cloud. The dataset is hosted both on AWS S3 and Azure Blob storage. 

Accessing a file via S3 using
```python
import fsspec
import xarray as xr

url = 's3://noaa-goes16/ABI-L2-MCMIPF/2020/002/00/OR_ABI-L2-MCMIPF-M6_G16_s20200020000216_e20200020009524_c20200020010031.nc'

fs = fsspec.filesystem('s3', anon=True)
f=fs.open(url)
ds = xr.open_dataset(f)
```

Takes ~30 seconds.

However, when I attempt to access the same data file on Azure using:
```python
import fsspec
import xarray as xr

url = 'az://noaa-goes16/ABI-L2-MCMIPF/2020/002/00/OR_ABI-L2-MCMIPF-M6_G16_s20200020000216_e20200020009524_c20200020010031.nc'

fs = fsspec.filesystem('az', account_name='goes')
f=fs.open(url)
ds = xr.open_dataset(f)
```
It takes ~5 minutes.

Am I doing something wrong? Is there a way to optimize this?

**Comment by martindurant** [comment]
For sure, HDF's access pattern can be pretty erratic through the file, but I know of no reason that azure should be slower by such a big factor. Perhaps you can turn on some logging?

(this issue probably belongs on adlfs)

cc @TomAugspurger @hayesgb

**Comment by TomAugspurger** [comment]
Where are you accessing the data from? Accessing the `geoseuwest` from a machinein west europe gives ~40s.

**Comment by lsterzinger** [comment]
@TomAugspurger this is running on Planetary Computer so I'm accessing from West Europe. Were you using the same code that I posted or were you accessing it some other way?

@martindurant I'm not familiar with logging on python, what's the best way to do this that would be most informative?

**Comment by lsterzinger** [comment]
Should have mentioned this is fsspec 2021.06.0 and xarray 0.18.2

**Comment by TomAugspurger** [comment]
I changed it to the westeurope bucket

```
url = 'az://noaa-goes16/ABI-L2-MCMIPF/2020/002/00/OR_ABI-L2-MCMIPF-M6_G16_s20200020000216_e20200020009524_c20200020010031.nc'

fs = fsspec.filesystem('az', account_name='goeseuwest')
f=fs.open(url)
```

**Comment by lsterzinger** [comment]
Okay cool, I'll check this out tomorrow. A thing I was confused about was the meaning of `account_name=`. What should be going here? I initially put `anonymous` or something but then I saw it threw an error about connecting to `anonymous.blob.core.windows.net` so I replaced it with `goes` since that's what was in the example notebook for GOES-16 access.

**Comment by TomAugspurger** [comment]
`account_name` is the [storage account](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview) name. A storage account + container is kind of like a bucket on s3.

**Comment by hayesgb** [comment]
@lsterzinger -- Which version of adlfs are you using?  There was a bug that caused read past end of files that's fixed in 0.7.7 that might contribute to this issue as well.

**Comment by lsterzinger** [comment]
`adlfs` was at 0.7.6, upgrading to 0.7.7 brought the opening time for the file on Azure down to about ~2 min (though I ran it a few times and got times <1 min). I then switched the account to `goeseuwest` as @TomAugspurger suggested and that seems to have brought it down to just ~10 sec, which is faster than the performance I was seeing on the S3 access. 

Keeping `adlfs` at 0.7.6 and just changing to the `goeseuwest` account alone also saw significant speed improvement, at around 1 min, but upgrading adlfs helped even more. 

Thank you all for your help!
